yellow-paper/docs/decentralization/actors.md

```diff
@@ -0,0 +1,152 @@
+
+The exact nature of the service can vary and have not been decided yet, but it could be some form of Oblivious Message Retrieval (OMR) service. This is a service that allows you to retrieve a message from a database without revealing which message you are retrieving. 
+
+Assuming OMR, the light-node would send a request to the OMR service, which would then retrieve the data from an Aztec node and return it to the light-node. The light-node would then validate the data against the headers it has stored and return it to the original requestor (for example the PXE).
+
+:::info Note Discovery Issue
+The issue of how to discover which notes belong to you is not solved by the Aztec protocol currently, and we have a [Request For Proposals](https://forum.aztec.network/t/request-for-proposals-note-discovery-protocol/2584) on the matter.
+:::
+
+#### Full-Node
+Can satisfy any query on current data, but will have to retrieve historical membership proofs from a third part or recompute it based on snapshots and blocks.
+
+Can utilize the same protocol as the Light-node for retrieving data that it don't already have.
+
+#### Archive-Node
+With a fully synched archive node you can respond to any query using your own data - no third party reliance.
+
+### Private eXecution Environment (PXE)
+
+The Aztec PXE is required to do anything meaningful for an end-user. It is the interface between the user and the Aztec network and acts as a private enclave that runs on the end-user's machine.
+
+While it is responsible for executing contracts and building proofs for the user's transactions privately, it needs data from the node to do so. It can be connected to any of the above types of nodes, inheriting their assumptions. From the point of view of the PXE, it is connected to a node, and it does not really care which one.
+
+When requesting encrypted notes from the node the PXE will decrypt it and store the decrypted notes in its own database. This database is used to build proofs for the user's transactions. 
+
+From most end users points of view, the PXE will be very close to a wallet.
+
+### Bundling the PXE with an Aztec ultra-light node
+
+A deployment of the PXE could be bundled together with an aztec ultra-light node which in turn is connected to "some" Ethereum node. The ethereum node could simply be infura, and then the ultra-light node handles its connections to the OMR service separately. 
+
+This setup can be made very light-weight and easy to use for the end-user, who might not be aware that they are running a node at all.
+
+The reasoning that the ultra-light node should be bundled with an internal ultra-light node as default instead of simple an RPC endpoint is to avoid users "missing" the OMR services and leak private information.
```

Comment:
```suggestion
The reasoning that the ultra-light node should be bundled with an internal ultra-light node by default instead of simply using an RPC endpoint is to avoid users "missing" the OMR services and leak private information.
```